mmv1/products/edgecontainer/Cluster.yaml

```diff
@@ -0,0 +1,515 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Cluster'
+description: |
+  Cluster contains information about a Google Distributed Cloud Edge Kubernetes cluster.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create and manage clusters': 'https://cloud.google.com/distributed-cloud/edge/latest/docs/clusters'
+  api: 'https://cloud.google.com/distributed-cloud/edge/latest/docs/reference/container/rest/v1/projects.locations.clusters'
+base_url: 'projects/{{project}}/locations/{{location}}/clusters'
+self_link: 'projects/{{project}}/locations/{{location}}/clusters/{{name}}'
+create_url: 'projects/{{project}}/locations/{{location}}/clusters?cluster_id={{name}}'
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The location of the resource.
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    immutable: true
+    url_param_only: true
+    description: |
+      The GDCE cluster name.
+import_format: ["projects/{{project}}/locations/{{location}}/clusters/{{name}}"]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_cluster"
+    primary_resource_id: "default"
+    vars:
+      edgecontainer_cluster_name: "basic-cluster"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_cluster_with_maintenance_window"
+    primary_resource_id: "default"
+    vars:
+      edgecontainer_cluster_name: "cluster-with-maintenance"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "edgecontainer_local_control_plane_cluster"
+    primary_resource_id: "default"
+    skip_test: true
```

Comment:
Skip the local control plane cluster test as we only have limited machine resources.
Instead the local control plane cluster test will be tested in the node pool test.

Reply:
Would you mind making these comments in the yaml file directly above this line? That will make it easier for us to keep track of

Reply:
Done.